### PR TITLE
[Build] Re-instate Windows tests

### PR DESCRIPTION
**Pull request by fnRaihanKibria**
- Installing the test Python environment works

### CODE REVIEW

**Comment by feltech** [comment]
Looking good so far.  Would be great to update `.github/workflows/test.yml` to re-enable Windows in the `runs-on` matrix, then we should see Windows show up in the checks in this PR.  Getting a tick next to the Windows check will be a good signal that this PR is ready.

File: .github/workflows/test.yml
Comment (feltech) [inline .github/workflows/test.yml:68]: [fix] 
* Remove `*_SANITIZER_*` variables, they're unix-only.
* Would be good to line-wrap if possible.

File: src/openassetio-core/CMakeLists.txt
Comment (feltech) [reply src/openassetio-core/CMakeLists.txt:37]: When I looked at the CI output, it looked like this may already be done (I saw some reference in the command line to a .pdb).  So perhaps this TODO can simply be removed (should be evident from a local Windows build)?

**Review by feltech — COMMENTED** [review]
Good stuff! I think `USES_TERMINAL` is reasonable. Just needs a bit of tidying up and I think this PR is good to go.

File: .github/workflows/test.yml
Comment (fnRaihanKibria) [reply .github/workflows/test.yml]: Fixed with https://github.com/TheFoundryVisionmongers/OpenAssetIO/pull/244/commits/a7c407c8f47bad76a4174cb319c3d1d502208e90 and https://github.com/TheFoundryVisionmongers/OpenAssetIO/pull/244/commits/c0d664c202ca31e383ddf68784f0b6c48abed789

File: src/openassetio-core/CMakeLists.txt
Comment (fnRaihanKibria) [reply src/openassetio-core/CMakeLists.txt]: Installing pdbs with https://github.com/TheFoundryVisionmongers/OpenAssetIO/pull/244/commits/318b11dd272f399cf3745c1c40ca8ee603e13478

File: resources/build/conanfile.txt
Comment (fnRaihanKibria) [reply resources/build/conanfile.txt]: Fixed with https://github.com/TheFoundryVisionmongers/OpenAssetIO/pull/244/commits/a07581895f9b763895121701c4f3a5bf3fe0441f

File: .github/workflows/test.yml
Comment (fnRaihanKibria) [inline .github/workflows/test.yml:67]: This actually isn't great since the path depends on the runner version. This will be different for a new Windows/compiler version. I saw some people managed to store it inside the matrix but I don't know YAML well enough right now to fix it straight away.

File: .github/workflows/test.yml
Comment (fnRaihanKibria) [inline .github/workflows/test.yml:84]: Also the path string is repeated.

File: .github/workflows/test.yml
Comment (feltech) [reply .github/workflows/test.yml:67]: The matrix stuff and the reference to it, e.g. `${{ matrix.os }}`, isn't YAML syntax, it's Github Action specific, so I would refer you to their docs.

However, there is also a way in YAML to avoid repetition - anchors and aliases, e.g. see https://www.educative.io/blog/advanced-yaml-syntax-cheatsheet

Comment (feltech) [reply .github/workflows/test.yml:67]: Perhaps under the matrix we can add a new variable under `matrix:`, e.g.

```
console_preamble: 
  # Ubuntu
  - ""
  # Windows
  - call "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Auxiliary\Build\vcvars64.bat"
```
Then add a `${{ matrix.console_preamble }}` line to the `run` blocks.

Does seem a bit of a waste having to have a blank Ubuntu line.  Surely there's a way in Github Actions to set a common variable outside the matrix, but I haven't looked.

You could perhaps add even more matrix parameters to parametrise the cmake command-line differences, avoiding the need for `if: runner.os`.  But that might not be possible, or be more hassle than it's worth.

OTIO's Github Actions may be a good reference.

File: src/openassetio-core/CMakeLists.txt
Comment (feltech) [reply src/openassetio-core/CMakeLists.txt]: Awesome! Obviously this isn't tested by CI but looks reasonable to me if you're happy it works OK.

File: .github/workflows/test.yml
Comment (feltech) [reply .github/workflows/test.yml]: Interesting usage of `>`, I didn't realise that you can break up the line wrapping by adding a blank line, really nice.

File: .github/workflows/test.yml
Comment (foundrytom) [reply .github/workflows/test.yml:67]: Btw - it's totally fine if the windows setup is a bit "hard coded" at this point, we can't sink too much time into it at this point. Just having a working version is good for now, even if it breaks in a month or so.

**Comment by foundrytom** [comment]
[question] No need to address this now, but given the number of windows specific steps, would it perhaps make more sense just to have a separate windows `.yml` file, rather than trying to shoehorn it into the matrix?

**Comment by foundrytom** [comment]
Nice work @fnRaihanKibria! Great to have this working now :) don't wait for me to merge this once @feltech is happy.

**Comment by fnRaihanKibria** [comment]
> [question] No need to address this now, but given the number of windows specific steps in CI, would it perhaps make more sense just to have a separate windows `.yml` file, rather than trying to shoehorn it into the matrix?

We might be able to clean this up as we go along and learn more YAML/Github features.

**Comment by fnRaihanKibria** [comment]
@feltech Commits have been squashed.

**Review by feltech — APPROVED** [review]

**Pull request merged**
